### ISSUE TITLE
Issue encountered when running make

### ISSUE THREAD

**Issue by Usamasource** [r-wasm/webr-repo#26]
Dear maintainers,

I am encountering an issue when attempting to build the project with the make command.

This error occurs in the context of a package installation step, however, I've ensured that the packages I'm trying to install are available and correctly specified in the project.

I have not been able to find a solution for this problem and any advice or help on how to troubleshoot this would be highly appreciated. I am happy to provide more information if required.
Here are the details of my environment:

Operating System: Ubuntu 22.04.2 LTS
R Version: 4.3

Thank you in advance for your time and support.

Best regards,

Usama

Message error:

Processing 128 package(s).
! Using bundled GitHub PAT. Please add your own PAT using `gitcreds::gitcreds_set()`.
✔ Loading metadata database ... done
./Matrix/
Warning: invalid package ‘Matrix’
Error: ERROR: no packages specified
./cli/
Warning: invalid package ‘cli’
Error: ERROR: no packages specified
./data.table/
Warning: invalid package ‘data.table’
Error: ERROR: no packages specified
./fs/
Warning: invalid package ‘fs’
Error: ERROR: no packages specified
./glue/
Warning: invalid package ‘glue’
Error: ERROR: no packages specified
./httpuv/
Warning: invalid package ‘httpuv’
Error: ERROR: no packages specified
./later/
Warning: invalid package ‘later’
Error: ERROR: no packages specified
./mgcv/
Warning: invalid package ‘mgcv’
Error: ERROR: no packages specified
...
p4.c:7:10: fatal error: 'proj.h' file not found
    7 | #include <proj.h>
      |          ^~~~~~~~
1 error generated.
OGR_write.cpp:1:10: fatal error: 'cpl_string.h' file not found
    1 | #include <cpl_string.h>
      |          ^~~~~~~~~~~~~~
p4.c:7:10: fatal error: 'proj.h' file not found
    7 | #include <proj.h>
      |          ^~~~~~~~
1 error generated.
OGR_write.cpp:1:10: fatal error: 'cpl_string.h' file not found
    1 | #include <cpl_string.h>
      |          ^~~~~~~~~~~~~~
1 error generated.
./rgeos.h:7:10: fatal error: 'geos_c.h' file not found
    7 | #include <geos_c.h>
      |          ^~~~~~~~~~
1 error generated.
...

**Comment by averissimo** [comment]
You probably need to go back to `webr` directory and build wasm libraries with the optional ones by: `cd libs && make all`

It seems that by default `webr` doesn't build with optional wasm libraries _(cairo, geos, etc...)_ and has no flag to indicate it. _(I might be wrong though)_.

Once it builds, come back to `webr-repo` and try to build again.

**Comment by averissimo** [comment]
You can actually look at how the `webR` deploy GitHub action is orchestrated to understand how to build everything.

https://github.com/r-wasm/webr/blob/main/.github/workflows/deploy.yml

It's based on @georgestagg Docker container with flang, but all other steps are in that action. In particular, line 46 with the instructions that I commented above.

**Comment by Usamasource** [comment]
Hi @averissimo !

Thank you for your response!
I have followed the suggested steps to solve my problem. I appreciate your efforts to resolve these issues. However, I am encountering the additional errors listed below, although I believe the .h header issues have been resolved. 

Thank you in advance!

Errors:
./Matrix/
Warning: invalid package ‘Matrix’
Error: ERROR: no packages specified
./cli/
Warning: invalid package ‘cli’
Error: ERROR: no packages specified
./data.table/
Warning: invalid package ‘data.table’
Error: ERROR: no packages specified
./fs/
Warning: invalid package ‘fs’
Error: ERROR: no packages specified
./glue/
Warning: invalid package ‘glue’
Error: ERROR: no packages specified
./httpuv/
Warning: invalid package ‘httpuv’
Error: ERROR: no packages specified
./later/
Warning: invalid package ‘later’
Error: ERROR: no packages specified
./mgcv/
Warning: invalid package ‘mgcv’
Error: ERROR: no packages specified
...
mintrin.h:16:2: error: "SSE instruction set not enabled"
   16 | #error "SSE instruction set not enabled"
      |  ^
1 error generated.
ssl.c:187:3: error: call to undeclared function 'SSL_set_fd'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
  187 |   SSL_set_fd(ssl, sockfd);
      |   ^
ssl.c:187:3: note: did you mean 'SSL_get_fd'?
SSL_get_fd' declared here
 1635 | __owur int SSL_get_fd(const SSL *s);
      |            ^
1 error generated.

./shiny/
Warning: invalid package ‘shiny’
Error: ERROR: no packages specified
./svglite/
Warning: invalid package ‘svglite’
Error: ERROR: no packages specified
./testthat/
Warning: invalid package ‘testthat’
Error: ERROR: no packages specified
Warning message:
In install.packages(pkg) :
  installation of package ‘shiny’ had non-zero exit status
c

**Comment by averissimo** [comment]
@Usamasource as a sanity check, can you try using this [`repo-update.R`](https://github.com/Appsilon/webr-repo/blob/main/repo-update.R)?

The one in this repo doesn't work in Linux when unpacking the zipball from GitHub

**Comment by averissimo** [comment]
I hacked the `openssl` R package build problem in a non-pretty way, but I don't think the hack stands other than building it.

The hack: remove the [#include guards](http://en.wikipedia.org/wiki/Include_guard) on lines 1643 / 1647 of `webr/libs/build/openssl-openssl-3.1.0/build/include/openssl/ssl.h`

I can't stress this enough: this only allows it to compile, it most probably won't work!

@georgestagg can you advise on the best way to compile the openssl package?

In particular, how to get the `SSL_set_fd` function declared if the native library is being compiled with `--no-sock`

**Comment by Usamasource** [comment]
Dear @georgestagg and @averissimo,
 
I wanted to express my heartfelt gratitude for your invaluable assistance. Your responses have been immensely helpful in resolving the installation issues I encountered on Linux. Thanks to your guidance, I have successfully incorporated the necessary changes, and the installation now proceeds without any apparent problem.

Your contributions are tremendously valuable and greatly appreciated.

Thank you so much for your assistance. Your kindness and expertise are deeply appreciated. :)

Warm regards,

Usama
